{"text": "2024-05-07 12:50:22 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:01.953020", "seconds": 1.95302}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:22.573634+08:00", "timestamp": 1715057422.573634}}}
{"text": "2024-05-07 12:50:26 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:05.698575", "seconds": 5.698575}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:26.319189+08:00", "timestamp": 1715057426.319189}}}
{"text": "2024-05-07 12:50:26 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Orcs Must Die! 3', 'url': 'https://store.epicgames.com/en-US/p/combo-a7e03a'}\n", "record": {"elapsed": {"repr": "0:00:06.042113", "seconds": 6.042113}, "exception": null, "extra": {"title": "Orcs Must Die! 3", "url": "https://store.epicgames.com/en-US/p/combo-a7e03a"}, "file": {"name": "claim.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:26.662727+08:00", "timestamp": 1715057426.662727}}}
{"text": "2024-05-07 12:50:26 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Cat Quest II', 'url': 'https://store.epicgames.com/en-US/p/cat-quest-ii-9dbefc'}\n", "record": {"elapsed": {"repr": "0:00:06.042663", "seconds": 6.042663}, "exception": null, "extra": {"title": "Cat Quest II", "url": "https://store.epicgames.com/en-US/p/cat-quest-ii-9dbefc"}, "file": {"name": "claim.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:26.663277+08:00", "timestamp": 1715057426.663277}}}
{"text": "2024-05-07 12:50:31 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/combo-a7e03a'}\n", "record": {"elapsed": {"repr": "0:00:10.721006", "seconds": 10.721006}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/combo-a7e03a"}, "file": {"name": "agent.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:31.341620+08:00", "timestamp": 1715057431.34162}}}
{"text": "2024-05-07 12:50:42 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3784), thread 'MainThread' (139820345121664): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/HI-plus/HI-plus/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/HI-plus/HI-plus/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/HI-plus/HI-plus/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/HI-plus/HI-plus/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 88, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 47, in __call__\n    do = self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 314, in iter\n    return fut.result()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 50, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/HI-plus/HI-plus/epic/src/epic_games/agent.py\", line 310, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 18871, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 471, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 70, in _expect_impl\n    raise AssertionError(\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n\n", "record": {"elapsed": {"repr": "0:00:21.563214", "seconds": 21.563214}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/HI-plus/HI-plus/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3784), thread 'MainThread' (139820345121664):", "module": "claim", "name": "__main__", "process": {"id": 3784, "name": "MainProcess"}, "thread": {"id": 139820345121664, "name": "MainThread"}, "time": {"repr": "2024-05-07 12:50:42.183828+08:00", "timestamp": 1715057442.183828}}}
